muxers/mplex/src/io.rs

```diff
@@ -22,22 +22,42 @@ use bytes::Bytes;
 use crate::{MplexConfig, MaxBufferBehaviour};
 use crate::codec::{Codec, Frame, LocalStreamId, RemoteStreamId};
 use log::{debug, trace};
-use fnv::FnvHashMap;
 use futures::{prelude::*, ready, stream::Fuse};
 use futures::task::{AtomicWaker, ArcWake, waker_ref, WakerRef};
 use futures_codec::Framed;
+use nohash_hasher::{IntMap, IntSet};
 use parking_lot::Mutex;
 use smallvec::SmallVec;
 use std::collections::VecDeque;
-use std::{cmp, io, mem, sync::Arc, task::{Context, Poll, Waker}};
+use std::{cmp, fmt, io, mem, sync::Arc, task::{Context, Poll, Waker}};
 
 pub use std::io::{Result, Error, ErrorKind};
 
+/// A connection identifier.
+///
+/// Randomly generated and mainly intended to improve log output
+/// by scoping substream IDs to a connection.
+#[derive(Clone, Copy)]
+struct Id(u32);
```

Comment:
I find it a bit confusing that `Id` is an identifier for a (multiplexed) stream whereas `LocalStreamId` is an identifier for a substream. Would it not be more descriptive if we had `StreamId` and `SubstreamId` ( / `LocalSubstreamId`)?

Reply:
I'm not really happy with either naming and struggled with it. "Stream" is certainly an appropriate and general term that can be used in all these cases and what kind of stream it is depends on the context. Making that context completely unambiguous in the name is tricky. The relative prefix "sub" only seems to help because in the context of libp2p we are mainly concerned with connections with one level of multiplexing. The API of this module currently uses the terminology `Multiplexed::poll_open_stream` etc. because it is clear from the context that these are "sub" streams of a multiplexed stream. Admittedly in the documentation and logs I've added the "sub" prefix for consistency with the public `StreamMuxer` API and terminology. The `Multiplexed` API on the other hand is only internal. All that said, would you go along for now with `Id` -> `ConnectionId`?

Reply:
What do you think of https://github.com/libp2p/rust-libp2p/pull/1785/commits/f6bb3544128f961a23261a6fb71a56cd4504c901?

Reply:
> The Multiplexed API on the other hand is only internal. All that said, would you go along for now with Id -> ConnectionId?

Yes. I am in favor of `ConnectionId` as it mirrors the [`core` `ConnectionId`](https://github.com/libp2p/rust-libp2p/blob/master/core/src/connection/manager.rs#L77).